### PR TITLE
[13.0] [MIG] partner_create_by_vat [WIP]

### PR DESCRIPTION
**Pull request by feketemihai**
Migrate partner_create_by_vat from 8.0, the module was not needed until 11.0 because Odoo had base_vat_autocomplete.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Why this is now needed in v13? Isn't `partner_autocomplete` providing this?

About the name, I think it's a bit misleading, as it only serves for VIES. I think `partner_data_vies_populator` or similar will fit better, don't you think?

**Comment by feketemihai** [comment]
@pedrobaeza Yes, it is almost the same, but this is without credits. I will change the name. Thanks

File: partner_create_by_vat/models/res_partner.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        country = self.env["res.country"].search([("code", "ilike", vat_country)])
+        if country:
+            res["country_id"] = country[0].id
+        return res
+
+    @api.onchange("vat")
+    def vies_vat_change(self):
+        for partner in self:
+            if not partner.vat:
+                pass
```
Comment (jarroyomorales) [inline partner_create_by_vat/models/res_partner.py]: This should be `continue` instead of `pass`

**Review by jarroyomorales — COMMENTED** [review]
Just a comment, but tested locally and the feature is working as expected :+1:

File: partner_create_by_vat/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    @api.model
+    def _get_vies_data(self, vat):
+        res = {}
+        vat = vat.strip().upper()
+        vat_country, vat_number = self._split_vat(vat)
+        result = check_vies(vat)
+        # Raise error if partner is not listed on Vies
+        if result.name is None:
+            raise ValidationError(_("The partner is not listed on Vies " "Webservice."))
```
Comment (jarroyomorales) [inline partner_create_by_vat/models/res_partner.py]: Black separated the strings

File: partner_data_vies_populator/models/res_partner.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.model
+    def _get_vies_data(self, vat):
+        res = {}
+        vat = vat.strip().upper()
+        vat_country, vat_number = self._split_vat(vat)
+        result = check_vies(vat)
```
Comment (jarroyomorales) [inline partner_data_vies_populator/models/res_partner.py:24]: This is failing if the country code is not european. One possible solution to do this before `check_vies`.
```python
try:
    compacted_vat = compact(vat)
except InvalidComponent:
    # This not an european company, so we are not processing it
    return {}
```

Comment (jarroyomorales) [reply partner_data_vies_populator/models/res_partner.py:24]: `compact` and `InvalidComponent` are imported as `from stdnum.eu.vat import check_vies, compact, InvalidComponent`

Comment (feketemihai) [reply partner_data_vies_populator/models/res_partner.py:24]: Update it checking the country code in base europe group

File: partner_data_vies_populator/README.rst [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+#. Clone the branch 13.0 of the repository https://github.com/OCA/partner-contact
+#. Add the path to this repository in your configuration (addons-path)
+#. Update the module list
+#. Search for "Partner Data VIES Populator" in your addons
+#. install the module
+
+Usage
+=====
+
+When changing a partners VAT number, this module will try to fetch the partner data from VIES webservice, if available on VIES will udpate the name, address and country.
```
Comment (NL66278) [inline partner_data_vies_populator/README.rst:53]: s/udpate/update

File: partner_data_vies_populator/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Copyright (C) 2015 Forest and Biomass Romania
+# Copyright (C) 2020 NextERP Romania
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
```
Comment (NL66278) [inline partner_data_vies_populator/__manifest__.py]: Make license lines https:

File: partner_data_vies_populator/tests/__init__.py [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# ©  2015 Forest and Biomass Services Romania
```
Comment (NL66278) [inline partner_data_vies_populator/tests/__init__.py]: Do not use copyright sign.

Comment (NL66278) [reply partner_data_vies_populator/tests/__init__.py]: Actually there should be no copyright line at all in an init file.

File: partner_data_vies_populator/tests/test_partner_data_vies_populator.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
+# Copyright (C) 2015 Forest and Biomass Romania
+# Copyright (C) 2020 NextERP Romania
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests.common import Form, TransactionCase
+
+
+class TestPartnerCreatebyVAT(TransactionCase):
```
Comment (NL66278) [inline partner_data_vies_populator/tests/test_partner_data_vies_populator.py]: By should have an uppercase `B`.

File: partner_data_vies_populator/tests/test_partner_data_vies_populator.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    def setUp(self):
+        super(TestPartnerCreatebyVAT, self).setUp()
```
Comment (NL66278) [inline partner_data_vies_populator/tests/test_partner_data_vies_populator.py]: Just use super(), python3 super style.

**Review by NL66278 — CHANGES_REQUESTED** [review]
Nice addition bu please attent to comments.

**Comment by bosd** [comment]
@feketemihai Are you still working on this?

**Comment by feketemihai** [comment]
@bosd Hopefully i will have some time this week to check and fix the comments.

**Comment by feketemihai** [comment]
> Nice addition bu please attent to comments.

@NL66278 Please check it again, i fixed all comments.

File: partner_data_vies_populator/README.rst (hunk 4 above)
Comment (NL66278) [reply partner_data_vies_populator/README.rst:53]: I still see udpate in the code.

**Comment by NL66278** [comment]
@feketemihai Please squash your commits into three: the original one for the migration, one for the translations and one for the review comments.

**Comment by NL66278** [comment]
@feketemihai Do you need help with squashing the commits as requested?

**Comment by feketemihai** [comment]
@NL66278 I'm busy with other developments, if you have time and you could do it, i would appreciate it.

**Comment by NL66278** [comment]
Translation commits squashed here. Other commits were from different contributors or had a well specified purpose. https://github.com/OCA/partner-contact/pull/1103

**Pull request closed without merging**